_posts/en/newsletters/2019-12-18-newsletter.md

```diff
@@ -0,0 +1,275 @@
+             return (None, None)
+         if data[0] > 16:
+             return (None, None)
+    -    if data[0] == 0 and len(decoded) != 20 and len(decoded) != 32:
+    +    if len(decoded) != 20 and len(decoded) != 32:
+             return (None, None)
+         return (data[0], decoded)
+    ```
+
+    If you have any questions or concerns about this suggested change,
+    please reply to the mailing list post linked in the *news* section
+    below.
+
+## News
+
+- **LN implementations add multipath payment support:** after a
+  significant amount of discussion and development, all three LN
+  implementations tracked by Optech have now added basic support for
+  [multipath payments][topic multipath payments] (merges: [C-Lightning][mpp cl],
+  [Eclair][mpp eclair], [LND][mpp lnd]).  A multipath payment
+  consists of multiple LN payments routed over different paths which may
+  all be claimed at the same time by the receiver.  This provides a
+  major upgrade to the usability of LN by allowing users to spend or
+  receive funds in several of their channels in the same overall
+  payment.  This upgrade means that spenders in particular have much
+  less need to worry about how much balance they have in any particular
+  channel as they can send up to their full available balance across all
+  their channels (subject to other LN limits).
+
+- **Analysis of bech32 error detection:** Pieter Wuille sent an
+  [email][wuille bech32 post] to the Bitcoin-Dev mailing list following
+  up on the bech32 malleability concerns described in previous
+  newsletters ([#72][news72 bech32], [#74][news74 bech32], and
+  [#76][news76 bech32]) where it's possible to add or remove any number of `q`
+  characters immediately before a `p` character at the end of a bech32
+  string.  Wuille's [analysis][wuille bech32 analysis] shows that this
+  is the only exception to bech32's expected error detection properties
+  and that "changing one constant in bech32 would resolve this issue."
+
+    Wuille plans to amend [BIP173][] to describe the weakness, propose a
+    change to limit existing bech32 address uses to either 20 byte or 32
+    byte witness program payloads, and define a modified version of
+    bech32 with the alternative constant for non-Bitcoin uses and for a
+    potential future where we want witness programs that are not 20
+    bytes or 32 bytes.
+
+- **Proposed changes to bip-ctv:** Jeremy Rubin [suggested][rubin ctv
+  update] several changes to the proposed soft fork addition of an
+  `OP_CHECKTEMPLATEVERIFY` (CTV) opcode.  Most
+  notably, the changes will remove the restriction that template
+  used by CTV cannot be derived from other data via Bitcoin Script.
+  This change simplifies the change to the Script language as discussed
+  in [Newsletter #75][news75 ctv].  Neither this change nor any of the
+  other changes modifies CTV's behavior in a way that would
+  significantly affect previously-described usecases as far as we're
```

Comment:
- can remove the hyphen
- "change" is used six times in this paragraph and not for the same things